courses/migrations/0017_coursecertificate_finalgrade_finalgradeaudit_programcertificate.py

```diff
@@ -0,0 +1,78 @@
+# Generated by Django 2.2.3 on 2019-08-09 06:57
+
+from django.conf import settings
+import django.contrib.postgres.fields.jsonb
+import django.core.validators
+from django.db import migrations, models
+import django.db.models.deletion
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        migrations.swappable_dependency(settings.AUTH_USER_MODEL),
+        ('courses', '0016_enrollment_order'),
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name='FinalGrade',
```

Comment:
How is the `FinalGrade` supposed to tie in with a `CourseCertificate`? If it is not supposed to, how is the course duration calculated for the certificate?

Reply:
I expect the course duration to be based on `course.start_date` and `course.end_date`

I expect program duration to be based on the first `course.start_date` to the last `course.end_date`

Reply:
I'm guessing you mean `courserun.start_date` and `courserun.end_date` here.

Reply:
Grades are associated with courseruns.

Reply:
Yes, that's correct:`courserun` not `course`.